mozilla-release/toolkit/components/extensions/webrequest/WebRequest.jsm

```
@@ -737,8 +756,21 @@ HttpObserverManager = {
 
       if (isSecondaryTab) {
         gBrowser.removeTab(isSecondaryTab);
+      } else {
+        try {
```

@tsl143 Please add our comment tag with the browser version.

---

And also the same question about try/catch. Do you really need this?

---

For try catch, I am expecting always the `tab[0]` exist, in weirdest case or by any future release this may change.

---

as above

---

> For try catch, I am expecting always the `tab[0]` exist, in weirdest case or by any future release this may change.

Then why not to check just its' existence? 
if (tabs[0] != null) {
  // here goes your code if tabs[0] exists;
}

---

yeah, true, but I am just following try catch scheme here for different checks, so sticking to it.

---

Ok. Then how can we know about the error? It will be silently ignored and swallowed by catch.
Either throw it explicitly in catch or use that tabs[0] against null check please.
I really do not want to fall into situation similar to what we had in SessionStore again.

---

We have added console to catch block, for better error deduction.